ElementX/Sources/Screens/HomeScreen/HomeScreenModels.swift

```diff
@@ -175,3 +178,88 @@ struct HomeScreenRoom: Identifiable, Equatable {
                        isPlaceholder: true)
     }
 }
+
+enum RoomListFilter: Int, CaseIterable, Identifiable {
+    var id: Int {
+        rawValue
+    }
+    
+    case people
+    case rooms
+    case unreads
+    case favourites
+    case lowPriority
+    
+    var localisedName: String {
+        switch self {
+        case .people:
+            return L10n.commonFilterPeople
+        case .rooms:
+            return L10n.commonFilterRooms
+        case .unreads:
+            return L10n.commonFilterUnreads
+        case .favourites:
+            return L10n.commonFilterFavourites
+        case .lowPriority:
+            return L10n.commonFilterLowPriority
+        }
+    }
+    
+    var complementaryFilter: RoomListFilter? {
+        switch self {
+        case .people:
+            return .rooms
+        case .rooms:
+            return .people
+        case .unreads:
+            return nil
+        case .favourites:
+            return .lowPriority
+        case .lowPriority:
+            return .favourites
+        }
+    }
+}
+
+final class RoomListFiltersState: ObservableObject {
+    @Published private var enabledFilters: Set<RoomListFilter>
+    
+    init(enabledFilters: Set<RoomListFilter> = []) {
+        self.enabledFilters = enabledFilters
+    }
+    
+    var sortedEnabledFilters: [RoomListFilter] {
+        enabledFilters.sorted(by: { $0.rawValue < $1.rawValue })
+    }
+    
+    var sortedAvailableFilters: [RoomListFilter] {
+        var availableFilters = Set(RoomListFilter.allCases)
+        for filter in enabledFilters {
+            availableFilters.remove(filter)
+            if let complementaryFilter = filter.complementaryFilter {
+                availableFilters.remove(complementaryFilter)
+            }
+        }
+        return availableFilters.sorted(by: { $0.rawValue < $1.rawValue })
+    }
+    
+    var isEmpty: Bool {
+        enabledFilters.isEmpty
+    }
+    
+    func setIsEnabled(_ isEnabled: Bool, for filter: RoomListFilter) {
```

Comment:
Here we say `isEnabled`, but below we check for `isSelected`. Would be nice to use the same for both. Also might be slight nicer to change this to

```suggestion
    func set(_ filter: RoomListFilter, isEnabled: Bool) {
```

So from the outside it is `filter.set(.rooms, isEnabled: true)`